src/main/java/org/jabref/gui/search/GlobalSearchResultDialog.java

```diff
@@ -0,0 +1,126 @@
+package org.jabref.gui.search;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.swing.undo.UndoManager;
+
+import javafx.collections.ObservableList;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.DialogPane;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.externalfiletype.ExternalFileTypes;
+import org.jabref.gui.groups.GroupViewMode;
+import org.jabref.gui.maintable.MainTableColumnModel;
+import org.jabref.gui.maintable.MainTableColumnModel.Type;
+import org.jabref.gui.maintable.MainTableDataModel;
+import org.jabref.gui.maintable.columns.FieldColumn;
+import org.jabref.gui.maintable.columns.SpecialFieldColumn;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.search.SearchQuery;
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.UnknownField;
+import org.jabref.model.groups.GroupTreeNode;
+import org.jabref.model.search.matchers.MatcherSet;
+import org.jabref.model.search.matchers.MatcherSets;
+import org.jabref.preferences.PreferencesService;
+
+public class GlobalSearchResultDialog {
```

Comment:
Why not make this a real dialog like all the other dialogs we have? That is, derive from Dialog, have fxml etc

Reply:
In the making.

Reply:
in progress